src/main/java/blackjack/domain/card/Card.java

```diff
@@ -0,0 +1,55 @@
+package blackjack.domain.card;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Card {
+
+    private static final List<Card> CARDS_CACHE = createCards();
+
+    private final CardPattern pattern;
+    private final CardNumber number;
+
+    private Card(final CardPattern pattern, final CardNumber number) {
+        this.pattern = pattern;
+        this.number = number;
+    }
+
+    public static Card of(final CardPattern pattern, final CardNumber number) {
+        return CARDS_CACHE.stream()
+                .filter(card -> isEqualCard(pattern, number, card))
+                .findAny()
+                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 카드입니다."));
+    }
+
+    private static boolean isEqualCard(final CardPattern pattern, final CardNumber number, final Card card) {
+        return card.pattern == pattern && card.number == number;
+    }
+
+    private static List<Card> createCards() {
+        final List<CardPattern> patterns = CardPattern.cardPatterns();
+        return patterns.stream()
+                .map(Card::createPatternCards)
+                .flatMap(List::stream)
+                .collect(Collectors.toList());
+    }
```

Comment:
저도 이전 로또 미션에서는 static block을 사용해서 진행하는방식으로 했는데 이번 미션에서는 static block을 고려하지 않았던 이유는 생성에대한 복잡성 때문이에요.
물론 한번만 호출되는 부분이기는 하지만 static block의 경우 하나의 block에서만 사용하기 때문에 내부의 복잡성이 올라가면 올라갈수록 구현 난이도가 까다로워지고, 오히려 가독성조차도 해친다고 생각해요.
```
However, this form of initialization has limitations because of its simplicity. If initialization requires some logic (for example, error handling or a for loop to fill a complex array), simple assignment is inadequate.
```

결과적으로는 한번만 호출되기 때문에 static block을 고려할 수는 있지만 복잡성이 올라가는 코드라면 오히려 메소드로 빼서 생성에 대한 가독성을 올려주는게 차후 유지보수에도 좋다라는 생각이 들었습니다. 🤔